### PR TITLE
Upgrade to Go 1.22

### PR DESCRIPTION
**Pull request by matthew1001**
All FireFly Go modules are being upgraded to Go 1.22. This PR updates Firefly itself, along with miscellaneous docs references.

### CODE REVIEW

**Review by dwertent — APPROVED** [review]
LGTM

**Review by EnriqueL8 — COMMENTED** [review]
looks good just that docs doesn't make sense

File: doc-site/docs/releasenotes/1.3_migration_guide.md [hunk 1]
```diff
@@ -103,4 +103,4 @@ Prior to FireFly v1.3.0, when the FabConnect client indexed events submitted by
 
 ### Go version upgrade
 
-FireFly v1.3.0 now uses Go 1.21 across all modules.
+FireFly v1.3.2 now uses Go 1.22 across all modules.
```
[the remark this answers is not in the document]
Comment (matthew1001) [reply doc-site/docs/releasenotes/1.3_migration_guide.md]: I've reverted it under the latest commit, but given that it is a `1.3` migration guide perhaps it's worth having a sentence for both in the docs?

[the remark this answers is not in the document]
Comment (matthew1001) [reply doc-site/docs/releasenotes/1.3_migration_guide.md]: If you migrate straight to `1.3.2` the docs aren't going to give you useful/accurate information

**Review by EnriqueL8 — APPROVED** [review]
Looks good - codecov is doing some weird

**Pull request merged**